flow/project.py

```diff
@@ -3052,23 +3052,37 @@ def _add_parameters(status):
                 if aggregate_id.startswith("agg-"):
                     # TODO: Fill parameters with empty values (or shared values?)
                     raise ValueError("Cannot show parameters for aggregates.")
-                statepoint = self.open_job(id=aggregate_id).statepoint()
+                job = self.open_job(id=aggregate_id)
+                # Cache the job state point and document if used to render status parameters.
+                statepoint = None
+                document = None
 
                 def dotted_get(mapping, key):
```

Comment:
This implementation looks great. Thanks @bdice @rosecers!

I am a little confused about the difference between the use of a new `cache` object and this. Should we be relying on `signac`'s caching for internal flow functionality? I am just worried that if a concept other than document is brought in by `signac`, we'd have to add more conditions to keep things optimized.

Reply:
Sort of. This "cache" behavior here is needed because we don't want to _fill_ the cache (read the state point / document) unless a key is required from the state point / document. This is operating at a slightly higher level than the signac state point cache and know more information about the keys being requested, so it is reasonable for signac-flow to handle this directly.